File: content/snippets/python/interview_lcci/10/01_sorted_merge.py
from typing import List


def merge(a: List[int], m: int, b: List[int], n: int) -> None:
    i = m - 1
    j = n - 1
    k = m + n - 1
    while j >= 0:
        if i >= 0 and a[i] > b[j]:
            a[k] = a[i]
            i -= 1
        else:
            a[k] = b[j]
            j -= 1
        k -= 1


def main() -> None:
    a = [1, 2, 3, 0, 0, 0]
    merge(a, 3, [2, 5, 6], 3)
    print(a)


if __name__ == "__main__":
    main()

File: content/snippets/typescript/interview_lcci/10/01_sorted_merge.ts
function merge(a: number[], m: number, b: number[], n: number): void {
    let i = m - 1;
    let j = n - 1;
    let k = m + n - 1;
    while (j >= 0) {
        if (i >= 0 && a[i] > b[j]) {
            a[k--] = a[i--];
        } else {
            a[k--] = b[j--];
        }
    }
}

const mergedValues = [1, 2, 3, 0, 0, 0];
merge(mergedValues, 3, [2, 5, 6], 3);
console.log(mergedValues);
